pkg/kubelet/kubelet.go

```diff
@@ -2248,22 +2249,40 @@ func (kl *Kubelet) rejectPod(pod *api.Pod, reason, message string) {
 // otherwise there will inevitably be pod delete create loops. This will be fixed
 // once we can extract these predicates into a common library. (#12744)
 func (kl *Kubelet) canAdmitPod(pods []*api.Pod, pod *api.Pod) (bool, string, string) {
-	if hasHostPortConflicts(pods) {
-		return false, "HostPortConflict", "cannot start the pod due to host port conflict."
-	}
-	if !kl.matchesNodeSelector(pod) {
-		return false, "NodeSelectorMismatching", "cannot be started due to node selector mismatch"
-	}
-	cpu, memory := kl.hasInsufficientfFreeResources(pods)
-	if cpu {
-		return false, "InsufficientFreeCPU", "cannot start the pod due to insufficient free CPU."
-	} else if memory {
-		return false, "InsufficientFreeMemory", "cannot be started due to insufficient free memory"
-	}
+	node, _ := kl.kubeClient.Core().Nodes().Get(kl.nodeName)
```

Comment:
Wow, this is really bad. I didn't realize the Kubelet didn't have the node information. This makes it extremely hard to refactor the predicates so that they are shared between scheduler and kubelet admission control. The scheduler predicates are pretty much all written such that you assume you have a Node.